### DISCUSSION TITLE
Applying Force/Pressure parallel to boundary

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#23377] [Q&A Modules: Solid mechanics] [answered]
Apologies, I'm sure this question must have been asked before but I can't seem to find it on the discussions page. I am trying to do a really simple beam bending problem in 3D. This problem is shown in the picture below. The blue end is fixed, and the red end has a load applied parallel to the surface, pulling the end of the beam down in the negative Z direction. I can't seem to identify the correct boundary condition to use for the force. I have tried using the Pressure BC, but as the Z component of the normal is initially 0, I don't think it will work? I have tried using it having set the component/displacements manually and I can't seem to model the correct behavior. Below is my input file, it's just a simple linear mechanics problem. Any help is much appreciated:)

![beam](https://user-images.githubusercontent.com/54721825/217508439-7b05f680-6a51-443b-9536-16f55b62692b.png)

```

[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
[]

[Mesh]
  file = ../myBeam_3.e
  boundary_id = '1 2 3'
[]

[Modules/TensorMechanics/Master]
  [./block1]
    strain = SMALL
    add_variables = true
  [../]
[]

[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    lambda = 50
    shear_modulus = 50
  [../]
  [./stress]
    type = ComputeLinearElasticStress
  [../]
[]

[BCs]
  [./zero_disp_x]
    type = DirichletBC
    variable = disp_x
    boundary = 1
    value = 0.0
  [../]

  [./zero_disp_y]
    type = DirichletBC
    variable = disp_y
    boundary = 1
    value = 0.0
  [../]

  [./zero_disp_z]
    type = DirichletBC
    variable = disp_z
    boundary = 1
    value = 0.0
  [../]

# here lies the confusion
  [Pressure]
    [Push]
      factor = 0.1
      boundary = 2
    []
  []
[]

[Preconditioning]
  [./SMP]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  type = Steady

  solve_type = 'NEWTON'

  petsc_options = '-snes_ksp_ew'
  petsc_options_iname = '-pc_type -sub_pc_type -pc_asm_overlap -ksp_gmres_restart'
  petsc_options_value = 'asm lu 1 101'
[]

[Outputs]
  exodus = true
  #perf_graph = true
[]
```

**Comment by hugary1995** [comment, marked as answer]
Let's start from the simple case. If the deformation is small enough (or equivalently the shear traction is a conservative force in the sense that it does _not_ follow the deformation of the beam), the shear traction will always be in the (-y) direction, so all you need is a `NeumannBC` or its variants.

If the deformation is large enough such that the "the shear traction is always in the (-y) direction" assumption is no longer good, things get a little more complicated, but not too bad. You'll first need to figure out the tangential direction along the side of the beam, call it (t), then apply the traction as tau = f t, where tau is the traction vector, f is the magnitude of the force, and (t) is the unit vector pointing in the tangential direction. In the input file, you will then need several (2 in 2D, 3 in 3D) instances of the BC to account for contributions from different components of the traction vector tau.

However, the more complicated case requires some code development, and I don't see an existing object in the tensor mechanics module that can do it for you. If you decide to write a pull request to MOOSE, I'm happy to guide you along the way.

**Reply by lyyc199586** [reply]
I am in a similar situation... should I look at the code for Pressure BC?

**Reply by GiudGiud** [reply]
It won't hurt.
Which case is your simulation in?

You may want to open your own discussions topic btw, it will have greater visibility

**Reply by lyyc199586** [reply]
LOL, you'are right, let me open another discussion
